lib/temporal/workflow/state_manager.rb

```diff
@@ -123,6 +124,10 @@ def apply_event(event)
 
         case event.type
         when 'WORKFLOW_EXECUTION_STARTED'
+          unless event.attributes.search_attributes.nil?
+            search_attributes.merge!(from_payload_map(event.attributes.search_attributes&.indexed_fields || {}))
```

Comment:
nit: You don't need the `&`, and possibly not the `|| {}` depending on the default for `indexed_fields`

Reply:
I think this is necessary because Ruby GRPC treats set and unset properties differently: `{}` if set but `nil` if unset. This works differently than Java GRPC which always sets a default value (here, `{}`). I've seen some other places where this behavior crops up in temporal-ruby where it's a non-issue in the Java SDK.

Moreover, I've seen differences where the Temporal server will set the value but the test-server emulator in the Java SDK does not. Because of the Java SDK's behavior with defaults, the difference doesn't matter to it, but causes divergent behavior in temporal-ruby between real and test servers. Better to protect against these differences and even against breaking changes that could be made in Temporal server.